server/middleware_test.go

```diff
@@ -177,3 +177,77 @@ func TestNoCacheHandler(t *testing.T) {
 		t.Errorf("expected no-cache Expires header to be '%#v', got '%#v'", want, got)
 	}
 }
+
+func TestAppIDHandler(t *testing.T) {
+	r, err := http.NewRequest("GET", "", nil)
+	if err != nil {
+		t.Error("failed to create mock request", "err", err)
+	}
+	w := httptest.NewRecorder()
+
+	id := "flambe"
+	AppIDHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requestID := GetRequestID(r.Context())
+		// write the ID to the response body so we can test it on the recorder
+		_, _ = w.Write([]byte(requestID))
+		w.WriteHeader(http.StatusOK)
+	}), &MockIDer{sendThis: id}).ServeHTTP(w, r)
+
+	headVal, ok := w.Result().Header[RequestIDHeader]
+	if !ok {
+		t.Error("header value was not found")
+	}
+	if len(headVal) != 1 {
+		t.Error("expected one value in request ID header", "got", len(headVal))
+	}
+	if headVal[0] != id {
+		t.Error("unexpected value in request ID header", "got", headVal[0], "expected", id)
+	}
+	if w.Body.String() != id {
+		t.Error("unexpected value in body", "got", w.Body.String(), "expected", id)
+	}
+}
+
+func TestPipelineIDHandler(t *testing.T) {
+	tests := []struct {
+		prev, next, expected string
+	}{
+		{"", "roger", "roger"},
+		{"roger", "roderick", "roger|roderick"},
+		{"roger|roderick", "brian", "roger|roderick|brian"},
+	}
+
+	for _, test := range tests {
+		r, err := http.NewRequest("GET", "", nil)
```

Comment:
for table driven test stuff each test case should be in a t.run, ie:

 ```suggestion
func TestPipelineIDHandler(t *testing.T) {
	tests := []struct {
		desc, prev, next, expected string
	}{
		{"new", "", "roger", "roger"},
		{"existing", "roger", "roderick", "roger|roderick"},
		{"chained", "roger|roderick", "brian", "roger|roderick|brian"},
	}

	for _, test := range tests {
		t.Run(test.desc, func(t *testing.T) {
			r, err := http.NewRequest("GET", "", nil)
			...
		})
	}
}
```

Reply:
thanks! fixed.